.github/workflows/pkgdown.yaml

```diff
@@ -55,11 +55,37 @@ jobs:
       - name: Install package
         run: R CMD INSTALL .
 
-      - name: Build pkgdown site
+      - name: Build pkgdown site (PR)
+        if: github.event_name != 'push'
+        shell: Rscript {0}
+        run: |
+          pkgdown::build_site(new_process = FALSE)
+      # Must validate after. Otherwise files are saved and `pkgdown::build_site()` gets mad
+      - name: Validate all topics exist (PR)
+        if: github.event_name != 'push'
+        shell: Rscript {0}
+        run: |
+          pkgdown::build_reference_index()
+          stopifnot(length(warnings()) == 0)
+
+      - name: Build pkgdown site (push)
+        if: github.event_name == 'push'
         run: pkgdown::build_site(new_process = FALSE)
         shell: Rscript {0}
 
-      - name: Deploy to Netlify
+      - name: Git setup (push)
+        if: github.event_name == 'push'
+        run: |
+          git config --local user.email "actions@github.com"
+          git config --local user.name "GitHub Actions"
+
+      - name: Deploy package (push)
+        if: github.event_name == 'push'
+        run: pkgdown::deploy_to_branch(new_process = FALSE)
+        shell: Rscript {0}
```

Comment:
The Deploy to Netlify step handles deployment

```suggestion
```

Reply:
🤦